src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -614,36 +595,70 @@ describe("Decimal", () => {
     );
 
     describe("i18n", () => {
-      beforeAll(() => {
-        I18n.locale = "fr";
+      describe("es", () => {
+        const esProps = { locale: "es-ES" };
+
+        it.each([
+          ["1000.23", "1000,23"],
+          ["10000.00", "10.000,00"],
+          ["10000.23", "10.000,23"],
+          ["100000.00", "100.000,00"],
+          ["1000000.00", "1.000.000,00"],
+        ])(" format %s to %s", (a, b) => {
+          render({ value: a, ...esProps });
+          expect(value()).toBe(b);
+          expect(hiddenValue()).toBe(a);
+        });
       });
 
-      afterAll(() => {
-        I18n.locale = undefined;
+      describe("pt", () => {
+        const ptProps = { locale: "pt-PT" };
+
+        it.each([
+          ["1000.23", "1000,23"],
+          ["10000.00", "10\xa0000,00"],
+          ["10000.23", "10\xa0000,23"],
+          ["100000.00", "100\xa0000,00"],
+          ["1000000.00", "1\xa0000\xa0000,00"],
+        ])(" format %s to %s", (a, b) => {
+          render({ value: a, ...ptProps });
+          expect(value()).toBe(b);
+          expect(hiddenValue()).toBe(a);
+        });
+
+        it(" handles a value that has white-spaces", () => {
```

Comment:
```suggestion
        it("handles a value that has white-spaces", () => {
```

Please also add a test for 

1. the space being in wrong location.

e.g. `1 0000,00` it should correct it.

2. two spaces

e.g. `10 000` it should not format it.